modules/ml/src/test/java/org/apache/ignite/ml/math/distances/WeightedMinkowskiDistanceTest.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ignite.ml.math.distances;
+
+import java.util.Arrays;
+import java.util.Collection;
+import org.apache.ignite.ml.math.primitives.vector.Vector;
+import org.apache.ignite.ml.math.primitives.vector.impl.DenseVector;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(Parameterized.class)
+public class WeightedMinkowskiDistanceTest {
+  /** Precision. */
+  private static final double PRECISION = 0.01;
+
+  @Parameterized.Parameters(name = "{0}")
+  public static Collection<TestData> data() {
+    return Arrays.asList(
+        new TestData(
+            new double[] {1.0, 0.0, 0.0},
+            new double[] {0.0, 1.0, 0.0},
+            1,
+            new double[] {1, 1, 1},
+            2.0
+        ),
+        new TestData(
+            new double[] {1.0, 0.0, 0.0},
+            new double[] {0.0, 1.0, 0.0},
+            2,
+            new double[] {1, 1, 1},
+            1.41
+        ),
+        new TestData(
+            new double[] {1.0, 0.0, 0.0},
+            new double[] {0.0, 1.0, 0.0},
+            3,
+            new double[] {1, 1, 1},
+            1.25
+        )
+    );
+  }
+
+  private final TestData testData;
+
+  public WeightedMinkowskiDistanceTest(TestData testData) {
+    this.testData = testData;
+  }
+
+  @Test
+  public void test() {
```

Comment:
change the test name to test+<something>, because it could create problems with next migration to next versions of JUNIt framework

Reply:
I renamed my tests like `test<distanceName`.